File: Solutions to a Quadratic Equation/solutions.java
import java.util.Scanner;
public class Solutions
{
    public static void main(String[] args) 
      {
        int a, b, c;
        double root1, root2, d;
        Scanner s = new Scanner(System.in);
        System.out.println("Given quadratic equation:ax^2 + bx + c");
        System.out.print("Enter a:");
        a = s.nextInt();
        System.out.print("Enter b:");
        b = s.nextInt();
        System.out.print("Enter c:");
        c = s.nextInt();
        System.out.println("Given quadratic equation:"+a+"x^2 + "+b+"x + "+c);
        d = b * b - 4 * a * c;
        if(d > 0)
        {
            System.out.println("Roots are real and unequal");
            root1 = ( - b + Math.sqrt(d))/(2*a);
            root2 = (-b - Math.sqrt(d))/(2*a);
            System.out.println("First root is:"+root1);
            System.out.println("Second root is:"+root2);
        }
        else if(d == 0)
        {
            System.out.println("Roots are real and equal");
            root1 = (-b+Math.sqrt(d))/(2*a);
            System.out.println("Root:"+root1);
        }
        else
        {
            System.out.println("Roots are imaginary");
        }
    }
}

File: Solutions to a Quadratic Equation/solutions.py
def roots(a, b, c):
    discriminant = (b**2) - (4*a*c)

    if discriminant > 0:
        print("The roots are different")
        print("The roots of the equation are:")
        root1 = (-b + discriminant)/(2*a)
        root2 = (-b - discriminant)/(2*a)

        print(root1, root2)

    elif discriminant == 0:
        print("The roots are equal")
        print("The roots of the equation are:")
        root = -b/(2*a)
        print(root)

    else:
        print("The roots are imaginary")
        print("The roots of the equation are:")
        i_root1 = (complex(-b, discriminant))/(2*a)
        i_root2 = (complex(-b, -discriminant))/(2*a)
        print(i_root1, i_root2)


a = int(input("Enter Coefficient of x^2 "))
b = int(input("Enter Coefficient of x "))
c = int(input("Enter Constant"))

roots(a, b, c)